vendor/engines/sanger_sequencing/app/assets/javascripts/sanger_sequencing/vue_well_plate_app.coffee

```diff
@@ -1,30 +1,32 @@
 window.vue_sanger_sequencing_well_plate_app = {
   props: ["submissions"]
   data: ->
-    plate: SangerSequencing.WellPlateBuilder.rows()
     builder: new SangerSequencing.WellPlateBuilder
 
   ready: ->
     vueBus.$on "submission-added", @addSubmission
     new AjaxModal(".js--modal", ".js--submissionModal")
 
   methods:
-    handleCellClick: (cell) ->
-      # TODO: Remove
-      console.log "handleCellClick", cell
-
     addSubmission: (submissionId) ->
       @builder.addSubmission @findSubmission(submissionId)
 
-    sampleAtCell: (cellName) ->
-      @builder.sampleAtCell(cellName)
+    removeSubmission: (submissionId) ->
+      @builder.removeSubmission @findSubmission(submissionId)
 
     findSubmission: (submissionId) ->
       @submissions.filter((submission) =>
         submission.id == submissionId
       )[0]
 
-    hasNotBeenAdded: (submissionId) ->
-      !@builder.hasBeenAdded(@findSubmission(submissionId))
+    styleForSubmissionId: (submissionId) ->
+      @colorBuilder ||= new SangerSequencing.WellPlateColors(@builder)
```

Comment:
It might make sense to explicitly assign `@colorBuilder` in the `ready` hook so that this method becomes somewhat simpler.